config.yml

```diff
@@ -23,12 +23,13 @@ service:
 
 native_service_types:
   - mysql
+  - aws
 
 # some id
 #connector_id: '1'
 
 sources:
   mongodb: connectors.sources.mongo:MongoDataSource
-  s3: connectors.sources.aws:S3DataSource
+  aws: connectors.sources.aws:S3DataSource
```

Comment:
since we might have other connectors for AWS services, we should keep `s3` in the name

Reply:
I understand your point here but if we keep s3 as source name here then our ftest will fail becuase in rest of the code we are using key name as "aws", so either we have to change the name from aws to s3 in allover the code structure or we can directly do this a single source name change in config.yml to make it work. Let me know which one we have to opt.

Reply:
```
if we keep s3 as source name here then our ftest will fail becuase we are using folder name "aws" everywhere in the connector and in that case file/folder name and source name are not matching and throws keyerror for "s3" when running ftest,
```

I don't understand this. we have a `service_type` which is `s3` and then a FQN (Fully Qualified Name) that is another separate thing. when you use `connectors.sources.aws:S3DataSource` it will look for the `S3DataSource` class in the `connectors.sources.aws` namespace.

Currently `aws` is a file. If you make it a directory you can use this structure for example:

```
connectors/
   sources/
      aws/
          __init__.py
         s3.py 
```

and have the `S3DataSource` class in `s3.py`, and then in `__init__.py` you can import `S3DataSource`